test/framework/src/main/java/org/elasticsearch/test/rest/client/RestTestClient.java

```diff
@@ -69,35 +52,29 @@
  * Wraps a {@link RestClient} instance used to send the REST requests.
  * Holds the {@link RestSpec} used to translate api calls into REST calls
  */
-public class RestTestClient implements Closeable {
-
-    public static final String PROTOCOL = "protocol";
-    public static final String TRUSTSTORE_PATH = "truststore.path";
-    public static final String TRUSTSTORE_PASSWORD = "truststore.password";
-
+public class RestTestClient {
     private static final ESLogger logger = Loggers.getLogger(RestTestClient.class);
-    //query_string params that don't need to be declared in the spec, thay are supported by default
+    //query_string params that don't need to be declared in the spec, they are supported by default
     private static final Set<String> ALWAYS_ACCEPTED_QUERY_STRING_PARAMS = Sets.newHashSet("pretty", "source", "filter_path");
 
     private final RestSpec restSpec;
     private final RestClient restClient;
     private final Version esVersion;
 
-    public RestTestClient(RestSpec restSpec, Settings settings, URL[] urls) throws IOException {
-        assert urls.length > 0;
+    public RestTestClient(RestSpec restSpec, RestClient restClient, List<HttpHost> hosts) throws IOException {
+        assert hosts.size() > 0;
         this.restSpec = restSpec;
-        this.restClient = createRestClient(urls, settings);
-        this.esVersion = readAndCheckVersion(urls);
-        logger.info("REST client initialized {}, elasticsearch version: [{}]", urls, esVersion);
+        this.restClient = restClient;
+        this.esVersion = readAndCheckVersion(hosts);
     }
 
-    private Version readAndCheckVersion(URL[] urls) throws IOException {
+    private Version readAndCheckVersion(List<HttpHost> hosts) throws IOException {
```

Comment:
> we should rather take the lower version and lose the validation

Right. I think if we revisit this code we really need to revision the whole round robin assumption. It is true but too brittle for my tastes.